src/wazuh_db/wdb_agent.c

```diff
@@ -11,269 +11,295 @@
     } else {
         time(&date);
     }
 
-    if (wdb_prepare(wdb_global, sql, -1, &stmt, NULL)) {
-        mdebug1("SQLite: %s", sqlite3_errmsg(wdb_global));
-        return -1;
-    }
+    sqlite3_snprintf(sizeof(wdbquery), wdbquery,  global_db_queries[SQL_INSERT_AGENT], id, name, ip, register_ip, key, date, group);
+    result = wdbc_query_ex(&wdb_sock, wdbquery, wdboutput, sizeof(wdboutput));
 
-    sqlite3_bind_int(stmt, 1, id);
-    sqlite3_bind_text(stmt, 2, name, -1, NULL);
-
-    if (ip){
-        sqlite3_bind_text(stmt, 3, ip, -1, NULL);
-    }
-    else {
-        sqlite3_bind_null(stmt, 3);
-    }
-
-    if(register_ip){
-        sqlite3_bind_text(stmt, 4, register_ip, -1, NULL);
-    }
-    else{
-        sqlite3_bind_null(stmt, 4);
+    switch (result){
+        case OS_SUCCESS:
+            result = wdb_create_agent_db(id, name);
+            break;
+        case OS_INVALID:
+            mdebug1("GLobal DB Error in the response from socket");
+            mdebug2("Global DB SQL query: %s", wdbquery);
+            break;
+        default:
+            mdebug1("GLobal DB Cannot execute SQL query; err database %s/%s.db", WDB2_DIR, WDB2_GLOB_NAME);
+            mdebug2("Global DB SQL query: %s", wdbquery);
+            break;
     }
 
-    if (key)
-        sqlite3_bind_text(stmt, 5, key, -1, NULL);
-    else
-        sqlite3_bind_null(stmt, 5);
-
-    sqlite3_bind_int64(stmt, 6, (long) date);
-    sqlite3_bind_text(stmt, 7, group, -1, NULL);
-
-    result = wdb_step(stmt) == SQLITE_DONE ? wdb_create_agent_db(id, name) : -1;
-    sqlite3_finalize(stmt);
-
     return result;
 }
 
 /* Update agent name. It doesn't rename agent DB file. It opens and closes the DB. Returns 0 on success or -1 on error. */
 int wdb_update_agent_name(int id, const char *name) {
     int result = 0;
-    sqlite3_stmt *stmt;
+    char wdbquery[OS_BUFFER_SIZE] = "";
+    char wdboutput[OS_BUFFER_SIZE] = "";
+    int wdb_sock = -1;
 
-    if (wdb_open_global() < 0)
-        return -1;
+    sqlite3_snprintf(sizeof(wdbquery), wdbquery, global_db_queries[SQL_UPDATE_AGENT_NAME], name, id);
+    result = wdbc_query_ex(&wdb_sock, wdbquery, wdboutput, sizeof(wdboutput));
 
-    if (wdb_prepare(wdb_global, SQL_UPDATE_AGENT_NAME, -1, &stmt, NULL)) {
-        mdebug1("SQLite: %s", sqlite3_errmsg(wdb_global));
-        return -1;
+    switch (result){
+        case OS_SUCCESS:
+            break;
+        case OS_INVALID:
+            mdebug1("GLobal DB Error in the response from socket");
+            mdebug2("Global DB SQL query: %s", wdbquery);
+            break;
+        default:
+            mdebug1("GLobal DB Cannot execute SQL query; err database %s/%s.db", WDB2_DIR, WDB2_GLOB_NAME);
+            mdebug2("Global DB SQL query: %s", wdbquery);
+            break;
     }
 
-    sqlite3_bind_text(stmt, 1, name, -1, NULL);
-    sqlite3_bind_int(stmt, 2, id);
-
-    result = wdb_step(stmt) == SQLITE_DONE ? 0 : -1;
-    sqlite3_finalize(stmt);
-
     return result;
```

Comment:
There is something very important here. The original `wdb_insert_agent` method had `0` and `-1` as possible return values. In your implementation, you are assigning to `result` the return value of `wdbc_query_ex`. This function can return `0`, `-1`, or `-2` so, we should make sure that in every single place where `wdb_insert_agent` is called, the `-2` value will not cause any unexpected behavior and will be handled correctly.
The verification that I mentioned could be very tedious! So, probably the best approach in terms of effort would be to keep the same possible return values for all the methods in this code. In this case for example, should be enough making this within the `default` case:

```c
result = OS_INVALID;
```

Please, review all the methods and as I mentioned, make sure that the possible return values will be the same as the original implementation.

Reply:
Now the `result = OS_INVALID;` solves the `-2` return problem.

But like the previous methods returned **...number of affected rows on success...**, returning **OS_SUCCESS** is indeed an error. 
Now, the switch case **OS_SUCCESS** assigns `result = 1`

The only exception is ` wdb_update_agent_keepalive();` because ` save_controlmsg()` expects **OS_SUCCESS**